python/lsst/ts/ofc/ofc_data/ofc_data.py

```diff
@@ -350,122 +314,207 @@ def _configure_instrument(self, value):
+                self.config_dir
+                / "image_quality_weights"
+                / f"{instrument}_gaussian_quadrature_weights.yaml"
+            )
+
+            try:
+                with open(gq_weights_path) as fp:
+                    gq_weights = yaml.safe_load(fp)
+            except FileNotFoundError:
+                raise RuntimeError(
+                    "Could not read gaussian quadrature weights file "
+                    f"from instrument config directory: {gq_weights_path!s}. "
+                    "Check your instrument configuration directory integrity."
+                )
+
         # Read y2 file
-        y2_path = inst_config_dir / self.y2_filename
+        # -------------
+        y2_path = self.config_dir / "y2" / f"{instrument}{self.y2_filename_root}"
 
         self.log.debug(f"Configuring y2: {y2_path}")
 
         try:
             with open(y2_path) as fp:
-                y2_correction = np.array(yaml.safe_load(fp))
+                y2_correction = yaml.safe_load(fp)
         except FileNotFoundError:
             raise RuntimeError(
                 f"Could not read y2 file from instrument config directory: {y2_path!s}. "
                 "Check your instrument configuration directory integrity."
             )
 
-        # Read all intrinsic zernike coefficients data.
-        # Get filenames.
-        intrinsic_zk_pattern = f"{self.intrinsic_zk_filename_root}*.yaml"
-        izk_filenames = [
-            Path(_p) for _p in glob(str(inst_config_dir / intrinsic_zk_pattern))
-        ]
-
-        self.log.debug(f"Configuring instrisic zernikes: {len(izk_filenames)} files.")
+        # If the camera type is lsst read and set up
+        # y2_correction for the gaussian quadrature points
+        # ------------------------------------------------
+        if instrument == "lsst":
+            gq_y2_path = (
+                self.config_dir
+                / "y2"
+                / f"{instrument}_gaussian_quadrature{self.y2_filename_root}"
+            )
 
-        intrinsic_zk = dict()
+            self.log.debug(f"Configuring y2: {gq_y2_path}")
 
-        for filename in izk_filenames:
-            filter_name_regex = _INTRISICZN_FILTER_REGEX.match(filename.name)
-            if filter_name_regex is None:
-                RuntimeError(
-                    f"Could not parse filter for intrinsic zernike coefficient file {filename.name}. "
-                    "Either remove or rename the file so it does not match the pattern "
-                    f"{intrinsic_zk_pattern}."
+            try:
+                with open(gq_y2_path) as fp:
+                    gq_y2_correction = yaml.safe_load(fp)
+            except FileNotFoundError:
+                raise RuntimeError(
+                    "Could not read y2 gaussian quadrature file "
+                    f"from instrument config directory: {gq_y2_path!s}. "
+                    "Check your instrument configuration directory integrity."
```

Comment:
This part of logic had been used too many times as the duplication. Please consider to write a simple function and reuse it. And please add the `RuntimeError` to the doc string.

Reply:
Done!